lib/maths/common/CBayesianOptimisation.cc

```diff
@@ -250,20 +284,26 @@ double CBayesianOptimisation::evaluate1D(const TVector& Kinvf, double input, int
     input = (input - m_MinBoundary(dimension)) /
             (m_MaxBoundary(dimension) - m_MinBoundary(dimension));
     for (std::size_t i = 0; i < m_FunctionMeanValues.size(); ++i) {
-        TVector x{this->transformTo01(m_FunctionMeanValues[i].first)};
+        const TVector& x{m_FunctionMeanValues[i].first};
         sum += Kinvf(static_cast<int>(i)) *
                CTools::stableExp(-(CTools::pow2(m_KernelParameters[dimension + 1]) +
                                    MINIMUM_KERNEL_COORDINATE_DISTANCE_SCALE) *
                                  CTools::pow2(input - x(dimension))) *
                prodXt(x, dimension);
     }
-
-    // We rewrite theta_0^2 sum - f_0 as (theta + f) * (theta - f) where
-    // theta = theta_0 sum^(1/2) and f = f_0^(1/2) because it has better
-    // numerics.
-    double theta{m_KernelParameters(0) * std::sqrt(sum)};
-    double f{std::sqrt(this->anovaConstantFactor(Kinvf))};
-    return (theta + f) * (theta - f);
+    double f2{this->anovaConstantFactor(Kinvf)};
+
+    if (std::signbit(sum) == std::signbit(f2)) {
```

Comment:
Can you please add a comment, what otherwise can go wrong here?

Reply:
See [726ea78](https://github.com/elastic/ml-cpp/pull/2259/commits/726ea7879d39c20ab374e5b5dda35525747d39b0).